module-2/backend/week_one.md

```diff
@@ -5,30 +5,69 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+</ul>
+
 2. What is Sinatra?
+<p>Its a domain specific languang that allows for web application development</p>
+
 4. What is MVC?
+<p>MVC stands for Model View Controller</p>
+
 5. Why do we follow conventions when creating our actions/path names in our Sinatra routes?
+<p>Because its RESTful?</p>
 6. What types of variables are accessible in our view templates without explicitly passing them?
+<p>Instance variables</p>
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
   
   ```ruby
   get '/horses' do
+    @count = 1
+    @name = "Mr. Ed"
     erb :index
   end
   ```
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
 9. What's the purpose of ERB?
+<p>Its a file type that allows us to embed ruby within html</p>
 10. Why do I need a development AND test database?
+<p>Because we often want to test how our app behaves under specific conditions. Using a seperate database for testing allows us to set those conditions and test the response. If we want to see how our app responds to invalid database entries it makes more sense to set those conditions in a test.</p>
 11. What's responsive design?
+<p>A design that allows for changes to be made with little difficulty.</p>
 12. What is CRUD and why is it important?
+<p>CRUD stands for create read update and delete. It's elements express the four basic functions of a database and how they relate to HTTP requests and responses. </p>
 13. What does HTTP stand for? 
+<p>Hypertext transfer protocol</p>
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+<p> <%= %> is for interpolation of variables <% %> interpolates ruby logic.</p>
 15. What's an ORM?
+<p>Object Relational Mapper- allows us to talk in an object oriented fashion to a non-object based database</p>
 16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
+<p>Active Record</p>
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+
+| What user wants                      | Verb   | Path                    |
+|--------------------------------------|--------|-------------------------|
+| See all restaurants                  | get    | '/restaurants'          |
+| See one restaurant                   | get    | '/restaurants/:id'      |
+| See form to add new restaurant       | get    | '/restaurants/new'      |
+| Submits and creates new restaurant   | post   | '/restaurants'          |
+| See form to edit existing restaurant | get    | '/restaurants/:id/edit' |
+| Update restaurant                    | put    | '/restaurants/:id'      |
+| Delete restaurant                    | delete | '/restaurants/:id'      |
+

```

Comment:
:tada: Looks like everything is in order, and markdown tables are kind of the best!